### PR TITLE
feat: Improve logging throughout the Dynamo path

### PR DESCRIPTION
**Pull request by gs-olive**
# Description
- Add clear logging at the beginning and end of each phase of compilation
- Reword logging in certain locations for clarity

Addresses #2399 

## Type of change

- Documentation and logging improvement

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/utils.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 from typing import Any, Callable, Dict, Optional, Sequence, Union
 
 import torch
-import torch_tensorrt
 from torch_tensorrt._Device import Device
 from torch_tensorrt._Input import Input
 from torch_tensorrt.dynamo import CompilationSettings
 from torch_tensorrt.dynamo._defaults import PRECISION
 
+import torch_tensorrt
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/utils.py:13]: is this required ?

File: py/torch_tensorrt/dynamo/compile.py [hunk 2]
```diff
@@ -42,6 +41,8 @@
     to_torch_tensorrt_device,
 )
 
+import torch_tensorrt
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/compile.py:44]: is this required ?

**Review by peri044 — CHANGES_REQUESTED** [review]

File: py/torch_tensorrt/dynamo/compile.py (hunk 2 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/compile.py:44]: The import itself is used on the following lines:
https://github.com/pytorch/TensorRT/blob/8df34b8a1a1ad7b2b2574208b2126c6d0704e38d/py/torch_tensorrt/dynamo/compile.py#L100-L108
Regarding the import line change, it was auto-completed by the linter.

File: py/torch_tensorrt/dynamo/utils.py (hunk 1 above)
Comment (peri044) [reply py/torch_tensorrt/dynamo/utils.py:13]: never mind

**Review by peri044 — APPROVED** [review]
LGTM

File: py/torch_tensorrt/dynamo/_compiler.py [hunk 3]
```diff
@@ -244,9 +245,11 @@ def compile_module(
     # Partition module into components that can be TRT-accelerated
     fast_partitioner_failed = False
 
+    logger.info("Beginning TensorRT operator Partitioning Phase")
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/_compiler.py:248]: Don't need this message, one of the following messages will be sufficient

File: py/torch_tensorrt/dynamo/_compiler.py [hunk 4]
```diff
@@ -256,21 +259,27 @@ def compile_module(
...
     if not settings.use_fast_partitioner:
+        logger.info("Partitioning the graph via the global partitioner")
         partitioned_module = partitioning.global_partition(
             gm,
             verbose=settings.debug,
             min_block_size=settings.min_block_size,
             torch_executed_ops=settings.torch_executed_ops,
         )
 
+    logger.info(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/_compiler.py:278]: Similar here, Following messages convey the same idea

File: py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py [hunk 5]
```diff
@@ -256,6 +256,10 @@ def run_node(self, n: torch.fx.Node) -> torch.fx.Node:
         n.kwargs = kwargs
 
         # run the node
+        _LOGGER.debug(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py:259]: Can we pull out some more metadata here, like input names and types, output shape etc.?

File: py/torch_tensorrt/dynamo/utils.py [hunk 6]
```diff
@@ -183,9 +183,16 @@ def parse_complex_tensor_structs(
 
         return torchtrt_inputs_dict
 
+    elif isinstance(inputs, (torch.SymBool, torch.SymFloat, torch.SymInt)):
+        raise ValueError(
+            f"Detected Torch symbolic input type {type(inputs)} during input parsing. "
+            "Symbolic inputs are not currently allowed; please specify dynamic=False "
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/utils.py:189]: Should it be dynamic=None ?

File: py/torch_tensorrt/dynamo/lowering/passes/_aten_lowering_pass.py [hunk 7]
```diff
@@ -23,6 +24,7 @@
         fuse_prims_broadcast,
         replace_max_pool_with_indices,
         view_to_reshape,
+        propagate_shapes,
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/lowering/passes/_aten_lowering_pass.py:27]: Do we need this for any specific thing ? The shape info is already present in the graph (in node metadata). Can we re-use that ?

File: py/torch_tensorrt/dynamo/utils.py (hunk 6 above)
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/utils.py:189]: `dynamic=None` can still allow dynamic shapes in the compilation (but prefers static), but `torch.compile` does not support dynamic shapes at all, so I thought this should be `False` to avoid dynamic shapes altogether

**Comment by gs-olive** [comment]
Sample logs:
```
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Converting node mean (kind: aten.mean.dim, args: ('[RELU]-[aten_ops.relu.default]-[relu_16]_output <tensorrt.ITensor [shape=(8, 512, 7, 7), dtype=DataType.HALF]>', [-1, -2], True))
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Ran node mean with properties: Inputs: (relu_16: (8, 512, 7, 7)@torch.float16, [-1, -2], True) | Outputs: (mean: (8, 512, 1, 1)@torch.float16)
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Running node reshape_default, a call_function node with target aten.reshape.default in the TensorRT Interpreter
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Converting node reshape_default (kind: aten.reshape.default, args: ('[REDUCE]-[aten_ops.mean.dim]-[mean]_output <tensorrt.ITensor [shape=(8, 512, 1, 1), dtype=DataType.HALF]>', [8, 512]))
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Ran node reshape_default with properties: Inputs: (mean: (8, 512, 1, 1)@torch.float16, [8, 512]) | Outputs: (reshape_default: (8, 512)@torch.float16)
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Running node _param_constant61, a get_attr node with target _param_constant61 in the TensorRT Interpreter
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Ran node _param_constant61 with properties: Inputs: () | Outputs: (_param_constant61: (1000,)@float16)
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Running node _frozen_param0, a get_attr node with target _frozen_param0 in the TensorRT Interpreter
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Ran node _frozen_param0 with properties: Inputs: () | Outputs: (_frozen_param0: (512, 1000)@float16)
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Running node addmm, a call_function node with target aten.addmm.default in the TensorRT Interpreter
DEBUG:torch_tensorrt.dynamo.conversion._TRTInterpreter:Converting node addmm (kind: aten.addmm.default, args: ('<torch.Tensor as np.ndarray [shape=(1000,), dtype=float16]>', '[SHUFFLE]-[aten_ops.reshape.default]-[reshape_default]_output <tensorrt.ITensor [shape=(8, 512), dtype=DataType.HALF]>', '<torch.Tensor as np.ndarray [shape=(512, 1000), dtype=float16]>'))
```

File: py/torch_tensorrt/dynamo/conversion/converter_utils.py [hunk 8]
```diff
@@ -44,6 +45,63 @@ def get_node_name(node: torch.fx.Node) -> str:
...
+    # Format input tensors
+    metadata_string = "Inputs: ("
+
+    # For each input argument, format it accordingly
+    for arg in node.args:
+        if isinstance(arg, torch.fx.Node):
+            if arg.op == "get_attr":
+                shape, dtype = constant_mapping[str(arg)]
+                arg_repr = f"{shape}@{dtype}"
+            elif arg.meta.get("tensor_meta", False):
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/conversion/converter_utils.py:77]: Could you also check arg.meta["val"] in addition to `tensor_meta`?

File: py/torch_tensorrt/dynamo/conversion/converter_utils.py [hunk 9]
```diff
@@ -44,6 +45,63 @@ def get_node_name(node: torch.fx.Node) -> str:
...
+    metadata_string = (
+        metadata_string[:-2] if metadata_string[-1] != "(" else metadata_string
+    ) + ")"
+
+    # Format output tensors and arguments
+    metadata_string += " | Outputs: ("
+    if node.op == "get_attr":
+        shape, dtype = constant_mapping[str(node)]
+        node_repr = f"{shape}@{dtype}"
+    elif node.meta.get("tensor_meta", False):
```
Comment (peri044) [inline py/torch_tensorrt/dynamo/conversion/converter_utils.py:95]: same as above

**Review by peri044 — CHANGES_REQUESTED** [review]

**Review by peri044 — APPROVED** [review]
LGTM
